### PR TITLE
URL: parser failure consistency

### LINKED ISSUE #3074: Add XHR runner for urltestdata.json
Replace XMLHttpRequest/open-url-bogus.htm (and maybe some others?) with a new file under url/ that runs all of url/urltestdata.json and expects SyntaxError for `"failure": true` entries.

cc @hallvors

### PR DESCRIPTION
**Pull request by annevk**
Fixes #3074.

### CODE REVIEW

**Comment by annevk** [comment]
https://bugs.chromium.org/p/chromium/issues/detail?id=693553

**Comment by w3c-bots** [comment]
# Chrome (unstable channel) #
Testing web-platform-tests at revision dd80487520e81830b03f7d5a7eb4b554a3a97ea0
Using browser at version 58.0.3013.3 dev
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/4911/url/failure.html">/url/failure.html</a></summary>


|                              Subtest                              | Results |
|-------------------------------------------------------------------|---------|
|                                                                   | OK      |
| `sendBeacon(): http://# should throw`                             | FAIL    |
| `sendBeacon(): sc://[/ should throw`                              | FAIL    |
| `URL's href: http://a:b@/www.example.com should throw`            | FAIL    |
| `XHR: https://[0:1.290.0.0.0] should throw`                       | PASS    |
| `Location's href: https://[0:1.290.0.0.0] should throw`           | FAIL    |
| `Location's href: https://example.com%A0/ should throw`           | FAIL    |
| `XHR: http://? should throw`                                      | PASS    |
| `XHR: sc:// / should throw`                                       | FAIL    |
| `sendBeacon(): http:a:b@/www.example.com should throw`            | FAIL    |
| `XHR: http://# should throw`                                      | PASS    |
| `URL's href: https://[0:0:] should throw`                         | FAIL    |
| `URL's href: sc://:/ should throw`                                | FAIL    |
| `Location's href: http://# should throw`                          | FAIL    |
| `sendBeacon(): ftp://example.com%A0/ should throw`                | FAIL    |
| `Location's href: http://foo:-80/ should throw`                   | FAIL    |
| `XHR: https://� should throw`                                     | PASS    |
| `Location's href: https://[0:1.23.23] should throw`               | FAIL    |
| `XHR: sc://@/ should throw`                                       | FAIL    |
| `sendBeacon(): https://x x:12 should throw`                       | FAIL    |
| `sendBeacon(): http://@:www.example.com should throw`             | FAIL    |
| `XHR: http:@/www.example.com should throw`                        | FAIL    |
| `Location's href: http:/a:b@/www.example.com should throw`        | FAIL    |
| `sendBeacon(): http::@/www.example.com should throw`              | FAIL    |
| `Location's href: http::@/www.example.com should throw`           | FAIL    |
| `XHR: http://@/www.example.com should throw`                      | PASS    |
| `XHR: http:@:www.example.com should throw`                        | FAIL    |
| `URL's href: https://[0:1.00.0.0.0] should throw`                 | FAIL    |
| `URL's href: sc://]/ should throw`                                | FAIL    |
| `sendBeacon(): https://[0:1:2:3:4:5:6:7.0.0.0.1] should throw`    | FAIL    |
| `Location's href: http:/:@/www.example.com should throw`          | FAIL    |
| `sendBeacon(): sc://]/ should throw`                              | FAIL    |
| `sendBeacon(): https://%EF%BF%BD should throw`                    | FAIL    |
| `Location's href: http:a:b@/www.example.com should throw`         | FAIL    |
| `XHR: http::@/www.example.com should throw`                       | FAIL    |
| `XHR: http:/a:b@/www.example.com should throw`                    | FAIL    |
| `sendBeacon(): sc://@/ should throw`                              | FAIL    |
| `XHR: http://user:pass@/ should throw`                            | PASS    |
| `sendBeacon(): file://example%/ should throw`                     | FAIL    |
| `URL's href: ftp://example.com%80/ should throw`                  | FAIL    |
| `sendBeacon(): https://[0::0::0] should throw`                    | FAIL    |
| `URL's href: http://@:www.example.com should throw`               | FAIL    |
| `Location's href: sc://:/ should throw`                           | FAIL    |
| `URL's href: sc:// / should throw`                                | FAIL    |
| `URL's href: http:/a:b@/www.example.com should throw`             | FAIL    |
| `sendBeacon(): https:@/www.example.com should throw`              | FAIL    |
| `sendBeacon(): https://� should throw`                            | FAIL    |
| `XHR: https:@/www.example.com should throw`                       | PASS    |
| `sendBeacon(): http://foo:-80/ should throw`                      | FAIL    |
| `Location's href: http://a:b@/www.example.com should throw`       | FAIL    |
| `URL's href: http://user:pass@/ should throw`                     | FAIL    |
| `XHR: http://[www.google.com]/ should throw`                      | PASS    |
| `Location's href: https:@/www.example.com should throw`           | FAIL    |
| `URL's href: file://example:test/ should throw`                   | FAIL    |
| `URL's href: non-special://[:80/ should throw`                    | FAIL    |
| `URL's href: sc://:12/ should throw`                              | FAIL    |
| `Location's href: https://[0:1:2:3:4:5:6:7.0.0.0.1] should throw` | FAIL    |
| `URL's href: http:@:www.example.com should throw`                 | FAIL    |
| `XHR: ftp://example.com%A0/ should throw`                         | PASS    |
| `sendBeacon(): file://example:1/ should throw`                    | FAIL    |
| `Location's href: sc://[/ should throw`                           | FAIL    |
| `URL's href: https://%EF%BF%BD should throw`                      | FAIL    |
| `Location's href: http://user@/www.example.com should throw`      | FAIL    |
| `sendBeacon(): https://[0:.0] should throw`                       | FAIL    |
| `Location's href: ftp://example.com%80/ should throw`             | FAIL    |
| `sendBeacon(): sc://:12/ should throw`                            | FAIL    |
| `Location's href: sc://te@s:t@/ should throw`                     | FAIL    |
| `Location's href: http://@:www.example.com should throw`          | FAIL    |
| `XHR: sc://]/ should throw`                                       | FAIL    |
| `URL's href: http::@/www.example.com should throw`                | FAIL    |
| `sendBeacon(): http://user:pass@/ should throw`                   | FAIL    |
| `Location's href: sc://\/ should throw`                           | FAIL    |
| `URL's href: https://example.com%A0/ should throw`                | FAIL    |
| `XHR: file://[example]/ should throw`                             | PASS    |
| `XHR: http://@:www.example.com should throw`                      | PASS    |
| `sendBeacon(): sc://:/ should throw`                              | FAIL    |
| `sendBeacon(): http://a:b@/www.example.com should throw`          | FAIL    |
| `XHR: non-special://[:80/ should throw`                           | FAIL    |
| `URL's href: http:@/www.example.com should throw`                 | FAIL    |
| `sendBeacon(): https://[0:1.290.0.0.0] should throw`              | FAIL    |
| `sendBeacon(): http:/a:b@/www.example.com should throw`           | FAIL    |
| `URL's href: https://[0:.0] should throw`                         | FAIL    |
| `URL's href: https://example.com%80/ should throw`                | FAIL    |
| `URL's href: http:a:b@/www.example.com should throw`              | FAIL    |
| `XHR: http:/@:www.example.com should throw`                       | FAIL    |
| `XHR: file://example:test/ should throw`                          | PASS    |
| `sendBeacon(): http://[www.google.com]/ should throw`             | FAIL    |
| `URL's href: sc://@/ should throw`                                | FAIL    |
| `XHR: https://[0:1.23.23] should throw`                           | PASS    |
| `XHR: http:/:@/www.example.com should throw`                      | FAIL    |
| `Location's href: http://[www.google.com]/ should throw`          | FAIL    |
| `URL's href: http://# should throw`                               | FAIL    |
| `URL's href: http:/@/www.example.com should throw`                | FAIL    |
| `Location's href: https://[0::0::0] should throw`                 | FAIL    |
| `XHR: https://[0::0::0] should throw`                             | PASS    |
| `XHR: https://[0:1.00.0.0.0] should throw`                        | PASS    |
| `XHR: https://x x:12 should throw`                                | FAIL    |
| `URL's href: http:/:@/www.example.com should throw`               | FAIL    |
| `Location's href: non-special://[:80/ should throw`               | FAIL    |
| `URL's href: sc:/// should throw`                                 | FAIL    |
| `Location's href: http://? should throw`                          | FAIL    |
| `URL's href: https:@/www.example.com should throw`                | FAIL    |
| `URL's href: http://user@/www.example.com should throw`           | FAIL    |
| `XHR: http://a:b@/www.example.com should throw`                   | PASS    |
| `XHR: file://example:1/ should throw`                             | PASS    |
| `URL's href: https://[0:1.23.23] should throw`                    | FAIL    |
| `Location's href: http:/@:www.example.com should throw`           | FAIL    |
| `Location's href: https://[0:0:] should throw`                    | FAIL    |
| `Location's href: https://� should throw`                         | FAIL    |
| `sendBeacon(): https://[0:1.00.0.0.0] should throw`               | FAIL    |
| `sendBeacon(): file://[example]/ should throw`                    | FAIL    |
| `XHR: https://[0:.0] should throw`                                | PASS    |
| `URL's href: http://? should throw`                               | FAIL    |
| `XHR: sc:/// should throw`                                        | FAIL    |
| `URL's href: sc://[/ should throw`                                | FAIL    |
| `Location's href: sc:/// should throw`                            | FAIL    |
| `sendBeacon(): http:/:@/www.example.com should throw`             | FAIL    |
| `sendBeacon(): sc://\/ should throw`                              | FAIL    |
| `Location's href: https://[0:.0] should throw`                    | FAIL    |
| `sendBeacon(): http://? should throw`                             | FAIL    |
| `URL's href: https://[0::0::0] should throw`                      | FAIL    |
| `Location's href: https://%EF%BF%BD should throw`                 | FAIL    |
| `Location's href: http:@/www.example.com should throw`            | FAIL    |
| `XHR: http://foo:-80/ should throw`                               | PASS    |
| `URL's href: file://example:1/ should throw`                      | FAIL    |
| `sendBeacon(): http://@/www.example.com should throw`             | FAIL    |
| `Location's href: file://example:test/ should throw`              | FAIL    |
| `Loading data…`                                                   | PASS    |
| `sendBeacon(): http:/@:www.example.com should throw`              | FAIL    |
| `Location's href: ftp://example.com%A0/ should throw`             | FAIL    |
| `Location's href: https://x x:12 should throw`                    | FAIL    |
| `XHR: file://example%/ should throw`                              | PASS    |
| `sendBeacon(): https://[0:1.23.23] should throw`                  | FAIL    |
| `Location's href: https://[0:1.00.0.0.0] should throw`            | FAIL    |
| `XHR: https://example.com%A0/ should throw`                       | PASS    |
| `Location's href: https://example.com%80/ should throw`           | FAIL    |
| `XHR: https://%EF%BF%BD should throw`                             | PASS    |
| `Location's href: http://user:pass@/ should throw`                | FAIL    |
| `sendBeacon(): sc://te@s:t@/ should throw`                        | FAIL    |
| `Location's href: http:/@/www.example.com should throw`           | FAIL    |
| `URL's href: sc://\/ should throw`                                | FAIL    |
| `XHR: https://[0:1:2:3:4:5:6:7.0.0.0.1] should throw`             | PASS    |
| `sendBeacon(): http:/@/www.example.com should throw`              | FAIL    |
| `URL's href: http://foo:-80/ should throw`                        | FAIL    |
| `Location's href: file://example:1/ should throw`                 | FAIL    |
| `URL's href: https://x x:12 should throw`                         | FAIL    |
| `URL's href: http:/@:www.example.com should throw`                | FAIL    |
| `URL's href: file://[example]/ should throw`                      | FAIL    |
| `sendBeacon(): sc:// / should throw`                              | FAIL    |
| `XHR: http://user@/www.example.com should throw`                  | PASS    |
| `XHR: http:/@/www.example.com should throw`                       | FAIL    |
| `sendBeacon(): https://[0:0:] should throw`                       | FAIL    |
| `sendBeacon(): non-special://[:80/ should throw`                  | FAIL    |
| `XHR: sc://[/ should throw`                                       | FAIL    |
| `Location's href: sc://]/ should throw`                           | FAIL    |
| `Location's href: file://example%/ should throw`                  | FAIL    |
| `URL's href: http://[www.google.com]/ should throw`               | FAIL    |
| `sendBeacon(): ftp://example.com%80/ should throw`                | FAIL    |
| `URL's href: https://[0:1:2:3:4:5:6:7.0.0.0.1] should throw`      | FAIL    |
| `Location's href: sc://@/ should throw`                           | FAIL    |
| `Location's href: http:@:www.example.com should throw`            | FAIL    |
| `sendBeacon(): file://example:test/ should throw`                 | FAIL    |
| `sendBeacon(): sc:/// should throw`                               | FAIL    |
| `XHR: sc://:12/ should throw`                                     | FAIL    |
| `XHR: sc://te@s:t@/ should throw`                                 | FAIL    |
| `XHR: sc://\/ should throw`                                       | FAIL    |
| `URL's href: ftp://example.com%A0/ should throw`                  | FAIL    |
| `URL's href: https://[0:1.290.0.0.0] should throw`                | FAIL    |
| `XHR: sc://:/ should throw`                                       | FAIL    |
| `URL's href: file://example%/ should throw`                       | FAIL    |
| `URL's href: https://� should throw`                              | FAIL    |
| `Location's href: sc:// / should throw`                           | FAIL    |
| `URL's href: sc://te@s:t@/ should throw`                          | FAIL    |
| `XHR: ftp://example.com%80/ should throw`                         | PASS    |
| `Location's href: file://[example]/ should throw`                 | FAIL    |
| `XHR: https://[0:0:] should throw`                                | PASS    |
| `XHR: https://example.com%80/ should throw`                       | PASS    |
| `sendBeacon(): https://example.com%A0/ should throw`              | FAIL    |
| `XHR: http:a:b@/www.example.com should throw`                     | FAIL    |
| `sendBeacon(): http://user@/www.example.com should throw`         | FAIL    |
| `URL's href: http://@/www.example.com should throw`               | FAIL    |
| `sendBeacon(): http:@/www.example.com should throw`               | FAIL    |
| `sendBeacon(): https://example.com%80/ should throw`              | FAIL    |
| `sendBeacon(): http:@:www.example.com should throw`               | FAIL    |
| `Location's href: http://@/www.example.com should throw`          | FAIL    |
| `Location's href: sc://:12/ should throw`                         | FAIL    |

</details>
</details>

**Comment by w3c-bots** [comment]
# Firefox (nightly channel) #
Testing web-platform-tests at revision dd80487520e81830b03f7d5a7eb4b554a3a97ea0
Using browser at version BuildID 20170123125947; SourceStamp 36486fdc3813ef7943ae5b07b4128866d1938a6c
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/4911/url/failure.html">/url/failure.html</a></summary>


|                              Subtest                              | Results |
|-------------------------------------------------------------------|---------|
|                                                                   | OK      |
| `sendBeacon(): http://# should throw`                             | FAIL    |
| `URL's href: sc:/// should throw`                                | FAIL    |
| `sendBeacon(): sc://[/ should throw`                              | PASS    |
| `URL's href: http://a:b@/www.example.com should throw`            | PASS    |
| `XHR: https://[0:1.290.0.0.0] should throw`                       | PASS    |
| `Location's href: https://example.com%A0/ should throw`           | FAIL    |
| `XHR: http://? should throw`                                      | FAIL    |
| `XHR: sc:// / should throw`                                       | FAIL    |
| `sendBeacon(): http:a:b@/www.example.com should throw`            | FAIL    |
| `XHR: http://# should throw`                                      | FAIL    |
| `URL's href: https://[0:0:] should throw`                         | PASS    |
| `URL's href: sc://:/ should throw`                                | FAIL    |
| `Location's href: http://# should throw`                          | FAIL    |
| `sendBeacon(): ftp://example.com%A0/ should throw`                | PASS    |
| `Location's href: http://foo:-80/ should throw`                   | FAIL    |
| `XHR: https://� should throw`                                     | FAIL    |
| `Location's href: https://[0:1.23.23] should throw`               | FAIL    |
| `XHR: sc://@/ should throw`                                       | FAIL    |
| `sendBeacon(): https://x x:12 should throw`                       | PASS    |
| `sendBeacon(): http://@:www.example.com should throw`             | PASS    |
| `XHR: http:@/www.example.com should throw`                        | FAIL    |
| `Location's href: http:/a:b@/www.example.com should throw`        | FAIL    |
| `sendBeacon(): http::@/www.example.com should throw`              | FAIL    |
| `Location's href: http::@/www.example.com should throw`           | FAIL    |
| `XHR: http://@/www.example.com should throw`                      | PASS    |
| `XHR: http:@:www.example.com should throw`                        | FAIL    |
| `URL's href: https://[0:1.00.0.0.0] should throw`                 | PASS    |
| `URL's href: sc://]/ should throw`                                | FAIL    |
| `sendBeacon(): https://[0:1:2:3:4:5:6:7.0.0.0.1] should throw`    | PASS    |
| `Location's href: http:/:@/www.example.com should throw`          | FAIL    |
| `sendBeacon(): sc://]/ should throw`                              | PASS    |
| `sendBeacon(): https://%EF%BF%BD should throw`                    | FAIL    |
| `Location's href: http:a:b@/www.example.com should throw`         | FAIL    |
| `XHR: http::@/www.example.com should throw`                       | FAIL    |
| `XHR: http:/a:b@/www.example.com should throw`                    | FAIL    |
| `sendBeacon(): sc://@/ should throw`                              | PASS    |
| `XHR: http://user:pass@/ should throw`                            | PASS    |
| `sendBeacon(): file://example%/ should throw`                     | PASS    |
| `URL's href: ftp://example.com%80/ should throw`                  | PASS    |
| `sendBeacon(): https://[0::0::0] should throw`                    | PASS    |
| `URL's href: http://@:www.example.com should throw`               | PASS    |
| `Location's href: sc://:/ should throw`                           | FAIL    |
| `URL's href: sc:// / should throw`                                | FAIL    |
| `URL's href: http:/a:b@/www.example.com should throw`             | PASS    |
| `sendBeacon(): https:@/www.example.com should throw`              | PASS    |
| `sendBeacon(): https://� should throw`                            | FAIL    |
| `XHR: https:@/www.example.com should throw`                       | PASS    |
| `sendBeacon(): http://foo:-80/ should throw`                      | PASS    |
| `Location's href: http://a:b@/www.example.com should throw`       | FAIL    |
| `URL's href: http://user:pass@/ should throw`                     | PASS    |
| `XHR: http://[www.google.com]/ should throw`                      | PASS    |
| `Location's href: https:@/www.example.com should throw`           | FAIL    |
| `URL's href: file://example:test/ should throw`                   | FAIL    |
| `URL's href: non-special://[:80/ should throw`                    | FAIL    |
| `URL's href: sc://:12/ should throw`                              | FAIL    |
| `Location's href: https://[0:1:2:3:4:5:6:7.0.0.0.1] should throw` | FAIL    |
| `URL's href: http:@:www.example.com should throw`                 | PASS    |
| `XHR: ftp://example.com%A0/ should throw`                         | FAIL    |
| `sendBeacon(): sc:/// should throw`                              | PASS    |
| `sendBeacon(): file://example:1/ should throw`                    | PASS    |
| `Location's href: sc://[/ should throw`                           | FAIL    |
| `URL's href: https://%EF%BF%BD should throw`                      | FAIL    |
| `Location's href: http://user@/www.example.com should throw`      | FAIL    |
| `sendBeacon(): https://[0:.0] should throw`                       | PASS    |
| `Location's href: ftp://example.com%80/ should throw`             | FAIL    |
| `sendBeacon(): sc://:12/ should throw`                            | PASS    |
| `Location's href: sc://te@s:t@/ should throw`                     | FAIL    |
| `Location's href: http://@:www.example.com should throw`          | FAIL    |
| `XHR: sc://]/ should throw`                                       | FAIL    |
| `URL's href: http::@/www.example.com should throw`                | PASS    |
| `sendBeacon(): http://user:pass@/ should throw`                   | PASS    |
| `Location's href: sc://\/ should throw`                           | FAIL    |
| `URL's href: https://example.com%A0/ should throw`                | PASS    |
| `XHR: file://[example]/ should throw`                             | FAIL    |
| `XHR: http://@:www.example.com should throw`                      | PASS    |
| `sendBeacon(): sc://:/ should throw`                              | PASS    |
| `sendBeacon(): http://a:b@/www.example.com should throw`          | PASS    |
| `XHR: non-special://[:80/ should throw`                           | FAIL    |
| `URL's href: http:@/www.example.com should throw`                 | PASS    |
| `sendBeacon(): https://[0:1.290.0.0.0] should throw`              | PASS    |
| `sendBeacon(): http:/a:b@/www.example.com should throw`           | FAIL    |
| `URL's href: https://[0:.0] should throw`                         | PASS    |
| `URL's href: https://example.com%80/ should throw`                | PASS    |
| `URL's href: http:a:b@/www.example.com should throw`              | PASS    |
| `XHR: http:/@:www.example.com should throw`                       | FAIL    |
| `XHR: file://example:test/ should throw`                          | FAIL    |
| `sendBeacon(): http://[www.google.com]/ should throw`             | PASS    |
| `URL's href: sc://@/ should throw`                                | FAIL    |
| `XHR: https://[0:1.23.23] should throw`                           | PASS    |
| `XHR: http:/:@/www.example.com should throw`                      | FAIL    |
| `Location's href: sc:/// should throw`                           | FAIL    |
| `Location's href: http://[www.google.com]/ should throw`          | FAIL    |
| `URL's href: http://# should throw`                               | FAIL    |
| `URL's href: http:/@/www.example.com should throw`                | PASS    |
| `Location's href: https://[0::0::0] should throw`                 | FAIL    |
| `XHR: https://[0::0::0] should throw`                             | PASS    |
| `XHR: https://[0:1.00.0.0.0] should throw`                        | PASS    |
| `XHR: https://x x:12 should throw`                                | PASS    |
| `URL's href: http:/:@/www.example.com should throw`               | PASS    |
| `Location's href: non-special://[:80/ should throw`               | FAIL    |
| `sendBeacon(): sc://te@s:t@/ should throw`                        | PASS    |
| `Location's href: http://? should throw`                          | FAIL    |
| `URL's href: https:@/www.example.com should throw`                | PASS    |
| `URL's href: http://user@/www.example.com should throw`           | PASS    |
| `XHR: http://a:b@/www.example.com should throw`                   | PASS    |
| `XHR: file://example:1/ should throw`                             | FAIL    |
| `URL's href: https://[0:1.23.23] should throw`                    | PASS    |
| `Location's href: http:/@:www.example.com should throw`           | FAIL    |
| `Location's href: https://[0:0:] should throw`                    | FAIL    |
| `Location's href: https://� should throw`                         | FAIL    |
| `sendBeacon(): https://[0:1.00.0.0.0] should throw`               | PASS    |
| `sendBeacon(): file://[example]/ should throw`                    | PASS    |
| `XHR: https://[0:.0] should throw`                                | PASS    |
| `URL's href: http://? should throw`                               | FAIL    |
| `URL's href: sc://[/ should throw`                                | FAIL    |
| `XHR: sc://:12/ should throw`                                     | FAIL    |
| `sendBeacon(): http:/:@/www.example.com should throw`             | FAIL    |
| `sendBeacon(): sc://\/ should throw`                              | PASS    |
| `Location's href: https://[0:.0] should throw`                    | FAIL    |
| `sendBeacon(): http://? should throw`                             | FAIL    |
| `URL's href: https://[0::0::0] should throw`                      | PASS    |
| `Location's href: https://%EF%BF%BD should throw`                 | FAIL    |
| `Location's href: http:@/www.example.com should throw`            | FAIL    |
| `XHR: http://foo:-80/ should throw`                               | PASS    |
| `URL's href: file://example:1/ should throw`                      | FAIL    |
| `sendBeacon(): http://@/www.example.com should throw`             | PASS    |
| `Location's href: file://example:test/ should throw`              | FAIL    |
| `Loading data…`                                                   | PASS    |
| `sendBeacon(): http:/@:www.example.com should throw`              | FAIL    |
| `Location's href: ftp://example.com%A0/ should throw`             | FAIL    |
| `Location's href: https://x x:12 should throw`                    | FAIL    |
| `XHR: file://example%/ should throw`                              | FAIL    |
| `sendBeacon(): https://[0:1.23.23] should throw`                  | PASS    |
| `Location's href: https://[0:1.00.0.0.0] should throw`            | FAIL    |
| `XHR: https://example.com%A0/ should throw`                       | FAIL    |
| `Location's href: https://example.com%80/ should throw`           | FAIL    |
| `XHR: https://%EF%BF%BD should throw`                             | FAIL    |
| `Location's href: http://user:pass@/ should throw`                | FAIL    |
| `XHR: sc:/// should throw`                                       | FAIL    |
| `Location's href: http:/@/www.example.com should throw`           | FAIL    |
| `URL's href: sc://\/ should throw`                                | FAIL    |
| `XHR: https://[0:1:2:3:4:5:6:7.0.0.0.1] should throw`             | PASS    |
| `sendBeacon(): http:/@/www.example.com should throw`              | FAIL    |
| `URL's href: http://foo:-80/ should throw`                        | PASS    |
| `Location's href: file://example:1/ should throw`                 | FAIL    |
| `URL's href: https://x x:12 should throw`                         | PASS    |
| `URL's href: http:/@:www.example.com should throw`                | PASS    |
| `URL's href: file://[example]/ should throw`                      | FAIL    |
| `sendBeacon(): sc:// / should throw`                              | PASS    |
| `XHR: http://user@/www.example.com should throw`                  | PASS    |
| `XHR: http:/@/www.example.com should throw`                       | FAIL    |
| `sendBeacon(): https://[0:0:] should throw`                       | PASS    |
| `sendBeacon(): non-special://[:80/ should throw`                  | PASS    |
| `XHR: sc://[/ should throw`                                       | FAIL    |
| `Location's href: sc://]/ should throw`                           | FAIL    |
| `Location's href: file://example%/ should throw`                  | FAIL    |
| `URL's href: http://[www.google.com]/ should throw`               | PASS    |
| `sendBeacon(): ftp://example.com%80/ should throw`                | PASS    |
| `URL's href: https://[0:1:2:3:4:5:6:7.0.0.0.1] should throw`      | PASS    |
| `Location's href: sc://@/ should throw`                           | FAIL    |
| `Location's href: http:@:www.example.com should throw`            | FAIL    |
| `sendBeacon(): file://example:test/ should throw`                 | PASS    |
| `Location's href: https://[0:1.290.0.0.0] should throw`           | FAIL    |
| `XHR: sc://te@s:t@/ should throw`                                 | FAIL    |
| `XHR: sc://\/ should throw`                                       | FAIL    |
| `URL's href: ftp://example.com%A0/ should throw`                  | PASS    |
| `URL's href: https://[0:1.290.0.0.0] should throw`                | PASS    |
| `XHR: sc://:/ should throw`                                       | FAIL    |
| `URL's href: file://example%/ should throw`                       | FAIL    |
| `URL's href: https://� should throw`                              | FAIL    |
| `Location's href: sc:// / should throw`                           | FAIL    |
| `URL's href: sc://te@s:t@/ should throw`                          | FAIL    |
| `XHR: ftp://example.com%80/ should throw`                         | FAIL    |
| `Location's href: file://[example]/ should throw`                 | FAIL    |
| `XHR: https://[0:0:] should throw`                                | PASS    |
| `XHR: https://example.com%80/ should throw`                       | FAIL    |
| `sendBeacon(): https://example.com%A0/ should throw`              | PASS    |
| `XHR: http:a:b@/www.example.com should throw`                     | FAIL    |
| `sendBeacon(): http://user@/www.example.com should throw`         | PASS    |
| `URL's href: http://@/www.example.com should throw`               | PASS    |
| `sendBeacon(): http:@/www.example.com should throw`               | FAIL    |
| `sendBeacon(): https://example.com%80/ should throw`              | PASS    |
| `sendBeacon(): http:@:www.example.com should throw`               | FAIL    |
| `Location's href: http://@/www.example.com should throw`          | FAIL    |
| `Location's href: sc://:12/ should throw`                         | FAIL    |

</details>
</details>

File: XMLHttpRequest/open-url-bogus.htm [hunk 1]
```diff
@@ -1,25 +0,0 @@
...
-        test(function() {
-          var client = new XMLHttpRequest()
-          assert_throws("SyntaxError", function() { client.open("GET", url) })
-        }, document.title + " (" + url + ")")
-      }
-      url("//[")
-      url("ftp:")
-      url("http://a a/")
-      url("http:////////////")
-      url("http://u:p@/")
```
Comment (tyoshino) [inline XMLHttpRequest/open-url-bogus.htm]: keep these cases tested by moving them to urltestdata.json?

Comment (annevk) [inline XMLHttpRequest/open-url-bogus.htm]: Yeah, see #3018.

**Review by zcorpan — APPROVED** [review]

**Pull request merged**